cpp/oneapi/dal/algo/decision_forest/test/batch_classification.cpp

```diff
@@ -0,0 +1,527 @@
+            }
+        }
+
+        if (check_mask_flag(desc.get_error_metric_mode(),
+                            error_metric_mode::out_of_bag_error_per_observation)) {
+            SECTION("oob error per observation shape is expected") {
+                REQUIRE(result.get_oob_err_per_observation().has_data());
+                REQUIRE(result.get_oob_err_per_observation().get_row_count() ==
+                        data.get_row_count());
+                REQUIRE(result.get_oob_err_per_observation().get_column_count() == 1);
+            }
+        }
+
+        if (variable_importance_mode::none != desc.get_variable_importance_mode()) {
+            SECTION("variable improtance shape is expected") {
+                REQUIRE(result.get_var_importance().has_data());
+                REQUIRE(result.get_var_importance().get_row_count() == 1);
+                REQUIRE(result.get_var_importance().get_column_count() ==
+                        data.get_column_count() - 1);
+            }
+        }
+    }
+
+    void check_infer_shapes(const df::descriptor<Float, Method, Task>& desc,
+                            const te::dataframe& data,
+                            const df::infer_result<Task>& result) {
+        SECTION("infer labels shape is expected") {
+            REQUIRE(result.get_labels().get_row_count() == data.get_row_count());
+            REQUIRE(result.get_labels().get_column_count() == 1);
+        }
+    }
+
+    void check_var_importance_matches_required(const df::descriptor<Float, Method, Task>& desc,
+                                               const df::train_result<Task>& train_result,
+                                               const te::dataframe& var_imp_data,
+                                               const te::table_id& data_table_id,
+                                               double accuracy_threshold) {
+        if (variable_importance_mode::none != desc.get_variable_importance_mode()) {
+            SECTION("match of variable importance vs required one is expected") {
+                const auto required_var_imp = var_imp_data.get_table(data_table_id);
+                std::int64_t row_ind = 0;
+                switch (desc.get_variable_importance_mode()) {
+                    case variable_importance_mode::mda_raw: row_ind = 1; break;
+                    case variable_importance_mode::mda_scaled: row_ind = 2; break;
+                    default: row_ind = 0; break;
+                };
+
+                const auto var_imp_val =
+                    dal::row_accessor<const Float>(train_result.get_var_importance()).pull();
+                const auto required_var_imp_val =
+                    dal::row_accessor<const float>(required_var_imp).pull({ row_ind, row_ind + 1 });
+
+                for (std::int32_t i = 0; i < var_imp_val.get_count(); i++) {
+                    REQUIRE(((required_var_imp_val[i] - var_imp_val[i]) / required_var_imp_val[i]) <
```

Comment:
Is required_var_imp_val[i] always not equal to zero?

Reply:
check added